client/modules/IDE/components/Timer.js

```diff
@@ -0,0 +1,46 @@
+import React, { PropTypes } from 'react';
+import moment from 'moment';
+
+class Timer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.showSavedTime = this.showSavedTime.bind(this);
+  }
+
+  componentWillMount() {
+    if (this.interval) {
+      clearInterval(this.interval);
```

Comment:
instead of clearing the interval here, maybe it makes more sense to clear the interval in [componentWillUnmount](https://facebook.github.io/react/docs/react-component.html#componentwillunmount)?